cargo/lib/dependabot/cargo/update_checker/latest_version_finder.rb

```diff
@@ -97,13 +114,43 @@ def crates_listing
             **SharedHelpers.excon_defaults
           )
 
-          @crates_listing = JSON.parse(response.body)
-        rescue Excon::Error::Timeout
-          retrying ||= false
-          raise if retrying
+          JSON.parse(response.body)
+        end
 
-          retrying = true
-          sleep(rand(1.0..5.0)) && retry
+        def crates_listing_sparse
+          info = dependency.requirements.map { |r| r[:source] }.compact.first
+          index = info && info[:index] 
+          registry_name = info && info[:name]
+          raise "Registry uses sparse protocol but no index URI found" if index.nil?
+          raise "Registry uses sparse protocol but no registry name found" if registry_name.nil?
+
+          # Default request headers
+          hdrs = { "User-Agent" => "Dependabot (dependabot.com)" }
+
+          registry_token_var_name = "CARGO_REGISTRIES_#{registry_name.upcase}_TOKEN"
```

Comment:
Could do with a reference to the cargo docs?

Reply:
Good spot! I'll link https://doc.rust-lang.org/cargo/reference/environment-variables.html in a comment.